src/libcrun/status.c

```diff
@@ -238,6 +241,37 @@ libcrun_read_container_status (libcrun_container_status_t *status, const char *s
     const char *detached[] = { "detached", NULL };
     status->detached = YAJL_IS_TRUE (yajl_tree_get (tree, detached, yajl_t_true));
   }
+  {
+    const char *external[] = { "external_descriptors", NULL };
+    const unsigned char *buf = NULL;
+    yajl_gen gen = NULL;
+    size_t buf_len;
+
+    gen = yajl_gen_alloc (NULL);
+    if (gen == NULL)
+      return crun_make_error (err, errno, "yajl_gen_alloc");
+    yajl_gen_array_open (gen);
+
+    tmp = yajl_tree_get (tree, external, yajl_t_array);
+    if (tmp && YAJL_IS_ARRAY (tmp))
+      {
+        size_t len = tmp->u.array.len;
+        int i;
+        for (i = 0; i < len; ++i)
+          {
+            yajl_val s = tmp->u.array.values[i];
+            if (s && YAJL_IS_STRING (s))
+              yajl_gen_string (gen,
+                               YAJL_STR (YAJL_GET_STRING (s)),
+                               strlen (YAJL_GET_STRING (s)));
```

Comment:
the compiler is raising a warning here: https://travis-ci.org/github/containers/crun/jobs/675685580#L904-L905, even though the code seems correct.

Could we just workaround it by having a temporary `char *tmp = YAJL_GET_STRING (s)` ?

Reply:
fixed (I hope)